slick/src/main/scala/slick/jdbc/DerbyProfile.scala

```diff
@@ -234,7 +234,7 @@ trait DerbyProfile extends JdbcProfile {
       } else super.createIndex(idx)
     }
 
-    override def dropIfExistsPhase = dropPhase1 ++ dropPhase2
+    override def dropIfExistsPhase = dropPhase2 // Derby does not support "DROP IF EXISTS"
```

Comment:
Can you explain this change? What is in `dropPhase1` and what are the ramifications of silently skipping it?

Reply:
Hi nafg,

This is the core of the change I'm doing for all Derby, Oracle and SQLServer (where I noticed it). Without this the test I'm introducing fails (but already passes for the other engines).

Onto your question: phase1 drops foreign keys. I don't think it needs or should be executed by `dropIfExists` because their dropping is unconditional and will fail the entire operation if they, for some reason, don't exist.

Ramifications: I failed in convincing myself there are any (but I appreciate, being new to the codebase, there might be some). Here's the reasoning: the original idea of running both phases seemed to me was that the order of `dropIfExists` deletions in a set of schemas shouldn't matter (because we'd start by removing foreign keys). But DDs compose `dropIfExists`s as a whole (and not by their phases) so this is already not going to work anyway (and one has to `dropIfExists` in the correct order by peeling off tables that no other tables reference first).

I'm actually running into this bug in a CI code that tries to scrub the DB regularly. Alternative fix could be introducing a phase that conditionally drops foreign keys (i.e. if they exist, or stays silent if not).

Hope that clarifies it somewhat,

Ales